src/backend/distributed/transaction/backend_data.c

```diff
@@ -63,6 +64,16 @@ typedef struct BackendManagementShmemData
 	 */
 	pg_atomic_uint64 nextTransactionNumber;
 
+	/*
+	 * Total number of client backends that are authenticated.
+	 * We only care about activeClientBackendCounter when adaptive
+	 * connection management is enabled, otherwise always zero.
+	 *
+	 * Note that the counter does not consider any background workers
+	 * or such, it only counts client_backends.
+	 */
+	pg_atomic_uint32 activeClientBackendCounter;
```

Comment:
mention that this does not always show the exact number that Postgres knows (e.g., clients authenticated has already used one entry in `ProcGlobal->freeProcs`). So, this is an approximation, but probably a pretty reasonable approximation.